src/tx/penalty/transaction.ts

```diff
@@ -285,17 +322,39 @@ export async function applyPenaltyTX(
     operatorAccount = wrappedStates[operatorShardusAddress].data as WrappedEVMAccount
   }
 
-
+  const { isDuplicate, eventTime } = isDuplicatePenaltyTx(tx, nodeAccount)
+  if (isDuplicate) {
+    /* prettier-ignore */ if (logFlags.dapp_verbose) console.log(`Duplicate penaltyTX: , TxId: ${txId}, reportedNode ${tx.reportedNodePublickKey}, ${{lastPenaltyTime: nodeAccount.nodeAccountStats.lastPenaltyTime, eventTime}}`)
+    shardus.applyResponseSetFailed(
+      applyResponse,
+      `applyPenaltyTX failed isDuplicatePenaltyTX reportedNode: ${tx.reportedNodePublickKey}`
+    )
+    return
+  }
 
   //TODO should we check if it was already penalized?
   const penaltyAmount = getPenaltyForViolation(tx, nodeAccount.stakeLock)
   applyPenalty(nodeAccount, operatorAccount, penaltyAmount)
+  nodeAccount.nodeAccountStats.penaltyHistory.push({
+    type: tx.violationType,
+    amount: penaltyAmount,
+    timestamp: eventTime,
+  })
   if (tx.violationType === ViolationType.LeftNetworkEarly) {
```

Comment:
in addition to LeftNetworkEarly event check, I think we should add if nodeAccount.rewardStart exists` && nodeAccount.rewardStart > 0 `